main.go

```diff
@@ -187,8 +191,8 @@ func newTLSConfig(cfg config.HTTPS) *tls.Config {
 	return &tlsCfg
 }
 
-func listenAndServe(ln net.Listener, h http.Handler, cfg config.TimeoutCfg) error {
-	s := &http.Server{
+func createServer(ln net.Listener, h http.Handler, cfg config.TimeoutCfg) *http.Server {
```

Comment:
if we're changing that part let's be golang agnostic and therefore (new given it returns a pointer)
```suggestion
func newServer(ln net.Listener, h http.Handler, cfg config.TimeoutCfg) *http.Server {
```